source/common/network/connection_impl.h

```diff
@@ -248,7 +248,7 @@ class ConnectionImpl : public ConnectionImplBase, public TransportSocketCallback
   uint64_t last_write_buffer_size_{};
   Buffer::Instance* current_write_buffer_{};
   uint32_t read_disable_count_{0};
-  DetectedCloseType detected_close_type_{DetectedCloseType::Normal};
+  StreamInfo::DetectedCloseType detected_close_type_{StreamInfo::DetectedCloseType::Normal};
```

Comment:
Should this be set to not-closed? Otherwise, it's weird that logs will print "normal" when a connection opens for a new connection log?

Reply:
I see, this is code is not in the Envoy repo, but it makes sense to add logs at beginning. 

If the goal is to distinguish the normal close case and not close case, we will need to set normal when closing the socket with the `undefined` as default, which will touch a quite few `connection_impl` codes. Right now this attribute was only added for RST detection, up to you if you want to change it, but we should not log `detected_close_type_` for non-close point access log.